### PR TITLE
Add fedora-messaging consumer to the vagrant environment.

### PR DESCRIPTION
**Pull request by cverna**
This commit adds a fedora-messaging configuration file example and
start the fedora-messaging service.

Signed-off-by: Clement Verna <cverna@tutanota.com>

### CODE REVIEW

**Comment by abompard** [comment]
I think it'd be nice to keep the `print-messages` service, don't you think?

File: devel/ansible/roles/bodhi/files/bodhi.toml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+amqp_url = "amqp://"
+
+publish_exchange = "amq.topic"
+
+callback = "bodhi.server.consumers:messaging_callback"
+
+# The topic_prefix configuration value will add a prefix to the topics of every sent message.
+# This is used for migrating from fedmsg, and should not be used afterwards.
+topic_prefix = ""
```
Comment (abompard) [inline devel/ansible/roles/bodhi/files/bodhi.toml]: You should set `org.fedoraproject.dev` here if you want to subscribe to the messages with a prefix (see the `[[bindings]]` section below.

File: devel/ansible/roles/bodhi/files/bodhi.toml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+[tls]
+ca_cert = "/etc/pki/tls/certs/ca-bundle.crt"
+keyfile = "/my/client/key.pem"
+certfile = "/my/client/cert.pem"
+
+[client_properties]
+app = "Bodhi"
+
+[exchanges.custom_exchange]
```
Comment (abompard) [inline devel/ansible/roles/bodhi/files/bodhi.toml]: This section should be removed.

File: devel/ansible/roles/bodhi/files/bodhi.toml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+durable = true
+auto_delete = false
+exclusive = false
+arguments = {}
+
+[qos]
+prefetch_size = 0
+prefetch_count = 25
+
+[consumer_config]
```
Comment (abompard) [inline devel/ansible/roles/bodhi/files/bodhi.toml]: If this section is unused, let's remove it.

File: devel/ansible/roles/bodhi/tasks/main.yml [hunk 4]
```diff
@@ -153,7 +153,6 @@
   with_items:
       - bodhi.service
       - fedmsg-tail.service
-      - print-messages.service
```
Comment (abompard) [inline devel/ansible/roles/bodhi/tasks/main.yml]: I think it would be nice to keep this service, don't you think?

**Review by abompard — CHANGES_REQUESTED** [review]

File: devel/ansible/roles/bodhi/tasks/main.yml (hunk 4 above)
Comment (cverna) [reply devel/ansible/roles/bodhi/tasks/main.yml]: yes actually it does not cost much to leave it

File: devel/ansible/roles/bodhi/files/bodhi.toml (hunk 1 above)
Comment (cverna) [reply devel/ansible/roles/bodhi/files/bodhi.toml]: Done, do I need to change anything from the bindings (remove the org.fedoraproject.*.) ?

**Comment by bowlofeggs** [comment]
On Tue, 2019-03-19 at 08:57 -0700, Clement Verna wrote:
> Done, do I need to change anything from the bindings (remove the
> org.fedoraproject.*.) ?

I would be in favor of killing this leftover from fedmsg. I see the
switch to fedora-messaging as a nice time to reset all expectations
about Bodhi's messages, and we are releasing a major version (4.0.0),
which signals backwards incompatible changes. In 
https://github.com/fedora-infra/bodhi/pull/2842 I am working on
changing the schemas we are publishing. What better way to indicate
that things changed than to stop using this silly prefix? ☺

**Comment by abompard** [comment]
I agree that the topic prefix is silly but topics are actually hardcoded in a lot of places that you don't want to touch (like the java Jenkins plugin), and in other places too because the topic to subscribe to is a class attribute in fedmsg, not configuration. If we change that we have to go over all possible consumers and update them.
My plan was to move to fedora-messaging using the exact same topic, and then change the topics as part of the schema change (since publishing a different schema is basically publishing on a different topic).

File: devel/ansible/roles/bodhi/files/bodhi.toml (hunk 1 above)
Comment (abompard) [reply devel/ansible/roles/bodhi/files/bodhi.toml]: No, they need to match, so you either add the prefix in `topic_prefix` or remove it from the bindings. It should work as it is now.

**Comment by bowlofeggs** [comment]
On Tue, 2019-03-19 at 09:47 -0700, Aurélien Bompard wrote:
> I agree that the topic prefix is silly but topics are actually
> hardcoded in a lot of places that you don't want to touch (like the
> java Jenkins plugin), and in other places too because the topic to
> subscribe to is a class attribute in fedmsg, not configuration. If we
> change that we have to go over all possible consumers and update
> them.
> My plan was to move to fedora-messaging using the exact same topic,
> and then change the topics as part of the schema change (since
> publishing a different schema is basically publishing on a different
> topic).

Fair enough, +1

**Comment by cverna** [comment]
So are we happy to merge this ? Or does it need something else ?

**Review by abompard — APPROVED** [review]
Yep, looks good :-)

**Pull request merged**